Add intel tensorflow-mkl singularity definition files (#1)
IntelAI/examples: machinelearning/intel-tensorflow/README.md

# Tensorflow

[TensorFlow*](https://github.com/tensorflow/tensorflow) is a predominantly-used machine learning framework in the deep learning arena,
demanding efficient utilization of computational resources. In order to take full advantage of Intel® architecture and to extract maximum performance,
the TensorFlow framework has been optimized using Intel® Math Kernel Library for Deep Neural Networks (Intel® MKL-DNN) primitives, a popular performance library for deep learning applications

> Here we are providing you with all pre-setup in singularity which is based of Tensorflow MKL [Dockerfile.devel-mkl](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/Dockerfile.devel-mkl)

Reviewer: ```suggestion
Provided here is all the setup required for Singularity, based on the following Intel Optimized Tensorflow Dockerfiles:  [Dockerfile.devel-mkl](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/Dockerfile.devel-mkl)
```

and [Dockerfile.mkl](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/docker/Dockerfile.mkl).

# Singularity.mkl.def

> * This is a light weight definition file targeted for data scientists, to build container you need to download whl from the following URL and save in the same directory as this file.

Reviewer: ```suggestion
* This is a light-weight definition file targeted for data scientists. To build this container, download a python whl from the following URL and save in the same directory as this README file:
```

https://pypi.org/project/intel-tensorflow/1.12.0/#files.

> * Update TF_WHL with wheel name in `%post` section and read comments in `%files` section to uncomment lines as instructed.

Reviewer: ```suggestion
* Update the TF_WHL variable with actual wheel name in the `%post` section, and then read comments in  the `%files` section and uncomment lines as instructed.
```

## Build

> * To build immutable containers (Production recommended).

Reviewer: ```suggestion
* To build immutable containers (Production recommended):
```

```
    sudo singularity build <mysingularity_name>.sif Singularity.mkl.def
```

> * To build development containers, Where you can install packages and modified container as needed.

Reviewer: ```suggestion
* To build development containers in which you can install additional packages and modify the container:
```

```
    sudo singularity build --sandbox <mysingularity_name>/ Singularity.mkl.def
```

> ## Shell and Exec

Author: Oh yeah, as singularity new to most of the people. I thought sample commands will help them.  We use `shell` and `exec` commands to test `mkl` initially. 

Still do you want to remove?
